src/adapters/scene-fetcher.ts

```diff
@@ -194,14 +202,26 @@ export async function createSceneFetcherComponent(
     return namesResponse.elements.some((element) => element.name.toLowerCase() === nameToValidate)
   }
 
+  async function hasWorldStreamingPermission(authAddress: string, worldName: string): Promise<boolean> {
+    const permissionsOverWorld = await fetchOnWorldActionPermissions(worldName)
+    let hasStreamingPermission = false
+    if (permissionsOverWorld?.streaming.type === PermissionType.AllowList) {
+      hasStreamingPermission = permissionsOverWorld.streaming.wallets.includes(authAddress)
+    }
+
+    return hasStreamingPermission
```

Comment:
What about something like:

```suggestion
    const { streaming } = permissionsOverWorld || { streaming: {} }
    return streaming.type === PermissionType.AllowList && streaming.wallets?.includes(authAddress)
```